server/app.js

```
@@ -1,5 +1,55 @@
 const express = require('express');
+
 const app = express();
+const fs = require('fs').promises;
+
+const dataPathFile = './data.json';
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+// GET THE TICKETS FROM DATA.JSON FILTER BY SEARCH TEXT
+
+app.get('/api/tickets', async (req, res) => {
+  const tickets = await fs.readFile(dataPathFile, 'utf-8');
+  const jsTickets = JSON.parse(tickets);
+  if (req.query.searchText) {
+    const filterTickets = jsTickets.filter(
+      (ticket) => ticket.title.toLowerCase().includes(req.query.searchText.toLowerCase()),
+    );
+    res.send(filterTickets);
+  } else {
+    res.send(jsTickets);
+  }
+});
+
+// POST- CHANGE THE TICKET DONE PROPERTY TO TRUE BY ID
+
+app.post('/api/tickets/:ticketId/done', async (req, res) => {
+  const tickets = await fs.readFile(dataPathFile, 'utf-8');
+  const jsTickets = JSON.parse(tickets);
+  jsTickets.forEach((ticket, i) => {
+    if (ticket.id === req.params.ticketId) {
```

What if `ticketId` is not found? we should return an error to the client